### PR TITLE
Fix 6667: Constant folding for unsigned long divide operation

### PR DESCRIPTION
**Pull request by hseok-oh**
Rrelated issue: #6667

For correctness of constant folding in 32bit architecture (ARM32) that use helper function - unsigned long divide or modulo operation.

### CODE REVIEW

**Comment by hseok-oh** [comment]
I think it will not get side effect for 64bit architectures because these architectures does not use helper function for 64bit integer divide operation. 

@jkotas @dotnet/jit-contrib PTAL

**Comment by cmckinsey** [comment]
That's true for x64/arm64. But this issue should also be a problem for our x86 RyuJIT so thanks for finding. @briansull /cc @dotnet/jit-contrib

File: src/jit/valuenum.cpp [hunk 1]
```diff
@@ -7061,10 +7061,10 @@ VNFunc Compiler::fgValueNumberHelperMethVNFunc(CorInfoHelpFunc helpFunc)
             vnf = VNFunc(GT_MOD);
             break;
         case CORINFO_HELP_ULDIV:
-            vnf = VNFunc(GT_DIV);
+            vnf = VNFunc(GT_UDIV);
             break; // Is this the right thing?
         case CORINFO_HELP_ULMOD:
-            vnf = VNFunc(GT_MOD);
+            vnf = VNFunc(GT_UMOD);
```
Comment (briansull) [inline src/jit/valuenum.cpp]: You can also remove the two comments:
`// Is this the right thing?`

**Comment by briansull** [comment]
LGTM

**Comment by hseok-oh** [comment]
@dotnet-bot test Linux ARM Emulator Cross Debug Build please

**Comment by hseok-oh** [comment]
@briansull PTAL

**Comment by hseok-oh** [comment]
Closed by miss-click, and reopen.

**Comment by RussKeldorph** [comment]
LGTM

**Comment by briansull** [comment]
LGTM

**Pull request merged**